From 28dac3212dd32ca8fc99e6fe81bef5a0cf3a6450 Mon Sep 17 00:00:00 2001
From: "yinbo.zhu" <yinbo.zhu@nxp.com>
Date: Sun, 23 Apr 2017 19:14:43 +0800
Subject: [PATCH 288/767] usb: DWC3: fix usb cause crash issue

remove dwc3_frame_length_adjustment for fix crash issue.

Signed-off-by: yinbo.zhu <yinbo.zhu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/usb/dwc3/core.c | 28 ----------------------------
 1 file changed, 28 deletions(-)

diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index 0b6242389111..5b50984384e2 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -274,31 +274,6 @@ static const struct clk_bulk_data dwc3_core_clks[] = {
 	{ .id = "suspend" },
 };
 
-/*
- * dwc3_frame_length_adjustment - Adjusts frame length if required
- * @dwc3: Pointer to our controller context structure
- */
-static void dwc3_frame_length_adjustment(struct dwc3 *dwc)
-{
-	u32 reg;
-	u32 dft;
-
-	if (dwc->revision < DWC3_REVISION_250A)
-		return;
-
-	if (dwc->fladj == 0)
-		return;
-
-	reg = dwc3_readl(dwc->regs, DWC3_GFLADJ);
-	dft = reg & DWC3_GFLADJ_30MHZ_MASK;
-	if (!dev_WARN_ONCE(dwc->dev, dft == dwc->fladj,
-	    "request value same as default, ignoring\n")) {
-		reg &= ~DWC3_GFLADJ_30MHZ_MASK;
-		reg |= DWC3_GFLADJ_30MHZ_SDBND_SEL | dwc->fladj;
-		dwc3_writel(dwc->regs, DWC3_GFLADJ, reg);
-	}
-}
-
 /**
  * dwc3_free_one_event_buffer - Frees one event buffer
  * @dwc: Pointer to our controller context structure
@@ -927,9 +902,6 @@ static int dwc3_core_init(struct dwc3 *dwc)
 	if (ret)
 		goto err1;
 
-	/* Adjust Frame Length */
-	dwc3_frame_length_adjustment(dwc);
-
 	dwc3_set_soc_bus_cfg(dwc);
 
 	usb_phy_set_suspend(dwc->usb2_phy, 0);
-- 
2.17.0

